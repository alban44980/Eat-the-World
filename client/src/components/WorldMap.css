html {
    box-sizing: border-box;
  }

h1 {
    text-align: center;
    margin-top: 0.5rem;
    margin-bottom:0;
}

.leaflet-container {
    display:flex;
    justify-content: center;
    background-color:white;
    border-radius: 2rem;
    padding: 0.1%;
    width: 90vw;
    height:50vh;
}

.map-search-container {
    display:flex;
    flex-direction: column;
    align-items: center;
}

.instructions {
    text-align: center;
}

.buttons {
    display:flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    margin-bottom: 5vh;
    font-family: Arial, Helvetica, sans-serif;
    width: 80vw;
}

.view-button {
    display:flex;
    width: 80vw;
    border-radius: 1rem;
    padding:0.5rem;
    align-self: center;
    justify-content: center;
    font-size: medium;
    font-weight:bold;
    text-decoration: none;
}

.random-button {
    margin-top: 0.5rem;
    width: 80vw;
    border-radius: 1rem;
    padding:0.5rem;
    font-size: medium;
    font-weight:bold;
}

.countryResult {
    background-color: white;
    height: 8vh;
    width: 70vw;
    overflow: hidden;
    overflow-y: auto;
}

.countryResult::-webkit-scrollbar {
    display: none;
}

.search-box {
    border-radius: 1rem;
    width: 60vw;
    padding:0.5rem;
    box-shadow: 4px 4px 6px 0 rgba(216, 13, 13, 0.3),
    -4px -4px 6px 0 rgba(116, 125, 136, .2), 
inset -4px -4px 6px 0 rgba(255,255,255,.2),
inset 4px 4px 6px 0 rgba(0, 0, 0, .2);
}

.select-country {
    width: 90vw;
    text-align: center;
}

h2 {
    background-color: white;
    opacity:0.8;
    width:80vw;
    border-radius: 1rem;
}

.home-container {
    display: flex;
    flex-direction: column;
    align-items: center;
}


