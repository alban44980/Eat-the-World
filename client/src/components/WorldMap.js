import React from 'react';
import './WorldMap.css';
import mapData from './../data/countries.json';
import { MapContainer, GeoJSON } from 'react-leaflet';
import 'leaflet/dist/leaflet.css';
import { Link } from 'react-router-dom';

export default function WorldMap ({countrySelected, SetSelectedCountry}) {

console.log(mapData);

// map styling
const countryStyle = {
    fillColor: 'red',
    fillOpacity: 0.2,
    color: 'black',
    weight: 0.5
}

// when a country is clicked, change the colour
const onCountryClick = (event) => {
    event.target.setStyle(
        {fillColor: "blue"}
    );
};

// list selected country
const writeCountryName = (event) => {
    // console.log(event.target.feature.properties.ADMIN);
    SetSelectedCountry(event.target.feature.properties.ADMIN);

}

// pop up produced following every country click with it's info
const onEachCountry = (country, layer) => {
    const countryName = country.properties.ADMIN;
    //console.log(countryName);
    //pop-up country name when clicked
    layer.bindPopup(countryName); 
    // change country color to blue when clicked 
    layer.on({
        click: onCountryClick
    })
    // list selected country
    layer.on({
        click: writeCountryName
    })
}

    
        return (
        <div>
            <h1>Eat the World</h1>
            <div className="map-search-container">
                <h2 className="instructions" >Select a country and view their food!</h2>

                <MapContainer style={{ height: "60vh" }} zoom={1.5} center={[20, 100]} >
                    <GeoJSON style={countryStyle} data={mapData.features} onEachFeature={onEachCountry} />
                </MapContainer>

                <h2 className="select-country">{countrySelected}</h2>
                <form >
                    <input type="text"  />
                    <button type="submit" >Search</button>
                </form>

                <Link to="/countrypage">
                    <button>View Their Food!</button>
                </Link>

                <div className="nav-bar">
                    <Link to="/">
                        <div className="home">Home</div>
                    </Link>
                    <Link to="/countrypage">
                        <div className="countrypage">Country</div>
                    </Link>
                    <Link to="/dishinfo">
                        <div className="food">Food</div>
                    </Link>
                </div>

            </div>

        </div>
        );
    
}
