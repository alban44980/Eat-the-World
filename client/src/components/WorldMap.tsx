import React, { useEffect, useState } from 'react';
import './WorldMap.css';
// import mapData from '../data/countries.json';
import { MapContainer, GeoJSON } from 'react-leaflet';
import 'leaflet/dist/leaflet.css';
import { Link, useHistory } from 'react-router-dom';
import Navbar from './Navbar';
import { StringMappingType } from 'typescript';
import * as API from '../ApiService';

//Type for data ==> GeoJson.FeatureCollection

// const data: any = mapData;

export default function WorldMap({
  countrySelected,
  SetSelectedCountry,
}: {
  countrySelected: string;
  SetSelectedCountry: React.Dispatch<React.SetStateAction<string>>;
}) {
  useEffect(() => {
    API.getCountryData().then((data) => {
      console.log('FROM USE EFFECT: ', data);
      setData(data);
    });
  }, []);

  const [data, setData] = useState<any>([]);

  const [filteredCountries, SetFilteredCountries] = useState<string[]>([]);
  const [wordEntered, SetWordEntered] = useState<string>('');

  // filters list of countries
  const handleFilter = (event: React.ChangeEvent<HTMLInputElement>) => {
    const searchInput: string = event.target.value;
    SetWordEntered(searchInput);
    //Come back to the filter type down there
    const newFilter: string[] = data.filter((value: any) => {
      return value.properties.ADMIN.toLowerCase().includes(
        searchInput.toLowerCase()
      );
    });
    SetFilteredCountries(newFilter);
  };

  const randomCountry = () => {
    const featuresArray = data;
    const randomIndex = Math.floor(Math.random() * featuresArray.length);
    SetSelectedCountry(featuresArray[randomIndex].properties.ADMIN);
  };

  const clearSearchInput = () => {
    SetWordEntered('');
  };

  // map styling
  const countryStyle: {
    fillColor: string;
    fillOpacity: number;
    color: string;
    weight: number;
  } = {
    fillColor: 'red',
    fillOpacity: 0.2,
    color: 'black',
    weight: 0.5,
  };

  // when a country is clicked, change the colour
  const onCountryClick = (event: any) => {
    event.target.setStyle({ fillColor: 'blue' });
  };

  // list selected country
  const writeCountryName = (event: any) => {
    SetSelectedCountry(event.target.feature.properties.ADMIN);
  };

  // Pop up produced following every country click with it's info
  const onEachCountry = (country: any, layer: any) => {
    const countryName = country.properties.ADMIN;
    //pop-up country name when clicked
    layer.bindPopup(countryName);
    // change country color to blue when clicked
    layer.on({
      click: onCountryClick,
    });
    // list selected country
    layer.on({
      click: writeCountryName,
    });
  };

  const history = useHistory();

  return (
    <div className="home-container">
      <h1>Eat the World</h1>
      <div className="map-search-container">
        <h2 className="instructions">Select a country and view their food</h2>

        {data.length ? (
          <MapContainer zoom={1.5} center={[41.38, 2.16]}>
            <GeoJSON
              style={countryStyle}
              data={data}
              onEachFeature={onEachCountry}
            />
          </MapContainer>
        ) : (
          <div className="scene">
            <div className="sky">
              <div className="sky__cloud-group">
                <div className="sky__cloud">
                  <div className="sky__cloud--bubbles"></div>
                </div>
                <div className="sky__cloud">
                  <div className="sky__cloud--bubbles"></div>
                </div>
                <div className="sky__cloud sky__cloud--small">
                  <div className="sky__cloud--bubbles"></div>
                </div>
                <div className="sky__cloud sky__cloud--small">
                  <div className="sky__cloud--bubbles"></div>
                </div>
                <div className="sky__cloud sky__cloud--small">
                  <div className="sky__cloud--bubbles"></div>
                </div>
              </div>
              <div className="sky__cloud-group">
                <div className="sky__cloud">
                  <div className="sky__cloud--bubbles"></div>
                </div>
                <div className="sky__cloud">
                  <div className="sky__cloud--bubbles"></div>
                </div>
                <div className="sky__cloud sky__cloud--small">
                  <div className="sky__cloud--bubbles"></div>
                </div>
                <div className="sky__cloud sky__cloud--small">
                  <div className="sky__cloud--bubbles"></div>
                </div>
                <div className="sky__cloud sky__cloud--small">
                  <div className="sky__cloud--bubbles"></div>
                </div>
              </div>
            </div>
            <div className="bird">
              <div className="bird__head">
                <div className="bird__head--hairs"></div>
                <div className="bird__head--eye"></div>
                <div className="bird__head--spot"></div>
                <div className="bird__head--beak"></div>
                <div className="bird__head--reflection">
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                </div>
              </div>
              <div className="bird__body"></div>
              <div className="bird__wing"></div>
              <div className="bird__legs">
                <div className="bird__leg bird__leg--left"></div>
                <div className="bird__leg bird__leg--right"></div>
              </div>
            </div>
          </div>
        )}

        <div
          onClick={() => history.push('/countrypage')}
          className="select-country"
        >
          <h2 className="select-country">{countrySelected + '!'}</h2>
        </div>

        <div className="buttons">
          <div className="search">
            <div className="searchInputs">
              <input
                className="search-box"
                type="text"
                placeholder="Search countries..."
                onChange={handleFilter}
                value={wordEntered}
              />
            </div>

            {filteredCountries.length !== 0 && (
              <div className="countryResult">
                {filteredCountries.slice(0, 10).map((country: any) => {
                  return (
                    <div
                      className="countrySuggestion"
                      onClick={() => {
                        SetSelectedCountry(country.properties.ADMIN);
                        SetFilteredCountries([]);
                        clearSearchInput();
                      }}
                    >
                      {country.properties.ADMIN}
                    </div>
                  );
                })}
              </div>
            )}
          </div>

          <div>
            <button
              className="random-button"
              onClick={() => {
                randomCountry();
                clearSearchInput();
                SetFilteredCountries([]);
              }}
            >
              {' '}
              Random!
            </button>
          </div>

          <Link to="/countrypage" className="view-button">
            <button className="view-button">View Their Food!</button>
          </Link>
        </div>
      </div>

      <Navbar />
    </div>
  );
}
