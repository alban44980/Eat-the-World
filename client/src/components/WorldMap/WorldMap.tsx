import React, { useEffect, useState } from 'react';
import './WorldMap.css';
import { MapContainer, GeoJSON } from 'react-leaflet';
import 'leaflet/dist/leaflet.css';
import { Link, useHistory } from 'react-router-dom';
import Navbar from './../NavBar/Navbar';
import { StringMappingType } from 'typescript';
import * as API from '../../ApiService';


export default function WorldMap({ countrySelected, SetSelectedCountry,}: {countrySelected: string; SetSelectedCountry: React.Dispatch<React.SetStateAction<string>>;}) {
  
  useEffect(() => {
    API.getCountryData().then((data) => {
      setData(data);
    });
  }, []);

  const [data, setData] = useState<any>([]);
  const [filteredCountries, SetFilteredCountries] = useState<string[]>([]);
  const [wordEntered, SetWordEntered] = useState<string>('');

  const handleFilter = (event: React.ChangeEvent<HTMLInputElement>) => {
    const searchInput: string = event.target.value;
    SetWordEntered(searchInput);
    const newFilter: string[] = data.filter((value: any) => {
      return value.properties.ADMIN.toLowerCase().includes(
        searchInput.toLowerCase()
      );
    });
    SetFilteredCountries(newFilter);
  };

  const randomCountry = () => {
    const featuresArray = data;
    const randomIndex = Math.floor(Math.random() * featuresArray.length);
    SetSelectedCountry(featuresArray[randomIndex].properties.ADMIN);
  };

  const clearSearchInput = () => {
    SetWordEntered('');
  };

  // map styling
  const countryStyle: {
    fillColor: string;
    fillOpacity: number;
    color: string;
    weight: number;
  } = {
    fillColor: 'red',
    fillOpacity: 0.2,
    color: 'black',
    weight: 0.5,
  };

  const onCountryClick = (event: any) => {
    event.target.setStyle({ fillColor: 'blue' });
  };

  const writeCountryName = (event: any) => {
    SetSelectedCountry(event.target.feature.properties.ADMIN);
  };

  const onEachCountry = (country: any, layer: any) => {
    const countryName = country.properties.ADMIN;
    layer.bindPopup(countryName);
    layer.on({
      click: onCountryClick,
    });
    layer.on({
      click: writeCountryName,
    });
  };

  const history = useHistory();

  return (
    <div className="home-container">
<<<<<<< HEAD:client/src/components/WorldMap/WorldMap.tsx
      <div className="titleContainer">
          <h1 className="title">Eat the World</h1>
      </div>
      <div className="map-container">
          {data.length ? (
            <MapContainer zoom={1.5} center={[41.38, 2.16]}>
              <GeoJSON
                style={countryStyle}
                data={data}
                onEachFeature={onEachCountry}
              />
            </MapContainer>
          ) : (
            <p>LOADING</p>
          )}
      </div>



      <div className="search-buttons-container">
        <div onClick={() => history.push('/countrypage')} className="select-country" >
            <h2 className="select-country-text"> {countrySelected} </h2>
=======
      <h1>Eat the World</h1>
      <div className="map-search-container">
        <h2 className="instructions">Select a country and view their food</h2>

        {data.length ? (
          <MapContainer zoom={1.5} center={[41.38, 2.16]}>
            <GeoJSON
              style={countryStyle}
              data={data}
              onEachFeature={onEachCountry}
            />
          </MapContainer>
        ) : (
          <div className="scene">
            <div className="sky">
              <div className="sky__cloud-group">
                <div className="sky__cloud">
                  <div className="sky__cloud--bubbles"></div>
                </div>
                <div className="sky__cloud">
                  <div className="sky__cloud--bubbles"></div>
                </div>
                <div className="sky__cloud sky__cloud--small">
                  <div className="sky__cloud--bubbles"></div>
                </div>
                <div className="sky__cloud sky__cloud--small">
                  <div className="sky__cloud--bubbles"></div>
                </div>
                <div className="sky__cloud sky__cloud--small">
                  <div className="sky__cloud--bubbles"></div>
                </div>
              </div>
              <div className="sky__cloud-group">
                <div className="sky__cloud">
                  <div className="sky__cloud--bubbles"></div>
                </div>
                <div className="sky__cloud">
                  <div className="sky__cloud--bubbles"></div>
                </div>
                <div className="sky__cloud sky__cloud--small">
                  <div className="sky__cloud--bubbles"></div>
                </div>
                <div className="sky__cloud sky__cloud--small">
                  <div className="sky__cloud--bubbles"></div>
                </div>
                <div className="sky__cloud sky__cloud--small">
                  <div className="sky__cloud--bubbles"></div>
                </div>
              </div>
            </div>
            <div className="bird">
              <div className="bird__head">
                <div className="bird__head--hairs"></div>
                <div className="bird__head--eye"></div>
                <div className="bird__head--spot"></div>
                <div className="bird__head--beak"></div>
                <div className="bird__head--reflection">
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                  <div className="bird__head--reflection--dot"></div>
                </div>
              </div>
              <div className="bird__body"></div>
              <div className="bird__wing"></div>
              <div className="bird__legs">
                <div className="bird__leg bird__leg--left"></div>
                <div className="bird__leg bird__leg--right"></div>
              </div>
            </div>
          </div>
        )}

        <div
          onClick={() => history.push('/countrypage')}
          className="select-country"
        >
          <h2 className="select-country">{countrySelected + '!'}</h2>
>>>>>>> alban-dev:client/src/components/WorldMap.tsx
        </div>



        <div className="search-container">
          <div className="search">

            <div className="searchInputs">
              <input
                className="search-box"
                type="text"
                placeholder="Search countries..."
                onChange={handleFilter}
                value={wordEntered}
              />
            </div>

            {filteredCountries.length !== 0 && (
              <div className="countryResult">
                {filteredCountries.slice(0, 10).map((country: any) => {
                  return (
                    <div
                      className="countrySuggestion"
                      onClick={() => {
                        SetSelectedCountry(country.properties.ADMIN);
                        SetFilteredCountries([]);
                        clearSearchInput();
                      }}
                    >
                      {country.properties.ADMIN}
                    </div>
                  );
                })}
              </div>
            )}

          </div>

          <Link to="/countrypage" className="view-button-container">
            <div className="view-button">
              <p className="button-label">View Food</p>
            </div>
          </Link>

        </div>

        <div>
          <button
            className="random-button"
            onClick={() => {
              randomCountry();
              clearSearchInput();
              SetFilteredCountries([]);
            }}
          >
            {' '}
            Random!
          </button>
        </div>


      </div>


    </div>
  );
}
