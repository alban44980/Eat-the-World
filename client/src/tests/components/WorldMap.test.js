import WorldMap from '../../components/WorldMap/WorldMap';
import { render, screen } from '@testing-library/react';

describe('Dish component', () => {
  test('Should match the spnapshot', () => {
    const { container } = render(<WorldMap />);
    expect(container.firstChild).toMatchSnapshot();
  });
  test('Should render the headings', () => {
    render(<WorldMap />);
    screen.getByText(/Eat the World/);
    screen.getByText(/View Food/);
    screen.getByText(/Random!/);
  });
});
