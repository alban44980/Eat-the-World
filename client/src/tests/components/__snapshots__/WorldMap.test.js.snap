// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dish component Should match the spnapshot 1`] = `
<div
  class="home-container"
>
  <div
    class="titleContainer"
  >
    <h1
      class="title"
    >
      Eat the World
    </h1>
  </div>
  <div
    class="map-container"
  >
    <div
      class="scene"
    >
      <div
        class="sky"
      >
        <div
          class="sky__cloud-group"
        >
          <div
            class="sky__cloud"
          >
            <div
              class="sky__cloud--bubbles"
            />
          </div>
          <div
            class="sky__cloud"
          >
            <div
              class="sky__cloud--bubbles"
            />
          </div>
          <div
            class="sky__cloud sky__cloud--small"
          >
            <div
              class="sky__cloud--bubbles"
            />
          </div>
          <div
            class="sky__cloud sky__cloud--small"
          >
            <div
              class="sky__cloud--bubbles"
            />
          </div>
          <div
            class="sky__cloud sky__cloud--small"
          >
            <div
              class="sky__cloud--bubbles"
            />
          </div>
        </div>
        <div
          class="sky__cloud-group"
        >
          <div
            class="sky__cloud"
          >
            <div
              class="sky__cloud--bubbles"
            />
          </div>
          <div
            class="sky__cloud"
          >
            <div
              class="sky__cloud--bubbles"
            />
          </div>
          <div
            class="sky__cloud sky__cloud--small"
          >
            <div
              class="sky__cloud--bubbles"
            />
          </div>
          <div
            class="sky__cloud sky__cloud--small"
          >
            <div
              class="sky__cloud--bubbles"
            />
          </div>
          <div
            class="sky__cloud sky__cloud--small"
          >
            <div
              class="sky__cloud--bubbles"
            />
          </div>
        </div>
      </div>
      <div
        class="bird"
      >
        <div
          class="bird__head"
        >
          <div
            class="bird__head--hairs"
          />
          <div
            class="bird__head--eye"
          />
          <div
            class="bird__head--spot"
          />
          <div
            class="bird__head--beak"
          />
          <div
            class="bird__head--reflection"
          >
            <div
              class="bird__head--reflection--dot"
            />
            <div
              class="bird__head--reflection--dot"
            />
            <div
              class="bird__head--reflection--dot"
            />
            <div
              class="bird__head--reflection--dot"
            />
            <div
              class="bird__head--reflection--dot"
            />
            <div
              class="bird__head--reflection--dot"
            />
            <div
              class="bird__head--reflection--dot"
            />
            <div
              class="bird__head--reflection--dot"
            />
            <div
              class="bird__head--reflection--dot"
            />
            <div
              class="bird__head--reflection--dot"
            />
            <div
              class="bird__head--reflection--dot"
            />
            <div
              class="bird__head--reflection--dot"
            />
            <div
              class="bird__head--reflection--dot"
            />
            <div
              class="bird__head--reflection--dot"
            />
            <div
              class="bird__head--reflection--dot"
            />
            <div
              class="bird__head--reflection--dot"
            />
            <div
              class="bird__head--reflection--dot"
            />
            <div
              class="bird__head--reflection--dot"
            />
            <div
              class="bird__head--reflection--dot"
            />
            <div
              class="bird__head--reflection--dot"
            />
          </div>
        </div>
        <div
          class="bird__body"
        />
        <div
          class="bird__wing"
        />
        <div
          class="bird__legs"
        >
          <div
            class="bird__leg bird__leg--left"
          />
          <div
            class="bird__leg bird__leg--right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="search-buttons-container"
  >
    <div
      class="select-country"
    >
      <h2
        class="select-country-text"
      >
         
         
      </h2>
    </div>
    <div
      class="search-container"
    >
      <div
        class="search"
      >
        <div
          class="searchInputs"
        >
          <input
            class="search-box"
            placeholder="Search countries..."
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="view-button-container"
      >
        <p
          class="button-label"
        >
          View Food
        </p>
      </div>
    </div>
    <div>
      <button
        class="random-button"
      >
         
        Random!
      </button>
    </div>
  </div>
</div>
`;
